src/handlers/predict.py

```diff
@@ -1,28 +1,104 @@
 from ..entities.prediction_request import PredictionRequestPydantic
 from ..helpers import model_decoder, json_to_predreq
+import base64
+import onnxruntime
+import numpy as np
+import torch
+import torch.nn.functional as F
+
+# import sys
+# import os
+
+# current_dir = os.path.dirname(__file__)
+# software_dir = os.path.abspath(os.path.join(current_dir, '../../../../../JQP'))
+# sys.path.append(software_dir)
+from jaqpotpy.descriptors.graph.graph_featurizer import SmilesGraphFeaturizer
 
 
 def model_post_handler(request: PredictionRequestPydantic):
-    model = model_decoder.decode(request.model['rawModel'])
+    model = model_decoder.decode(request.model["rawModel"])
     data_entry_all = json_to_predreq.decode(request, model)
     prediction = model.predict_onnx(data_entry_all)
-    if model.task == 'classification':
+    if model.task == "classification":
         probabilities = model.predict_proba_onnx(data_entry_all)
     else:
         probabilities = [None for _ in range(len(prediction))]
 
     results = {}
     for i, feature in enumerate(model.dependentFeatures):
-        key = feature['key']
+        key = feature["key"]
         if len(model.dependentFeatures) == 1:
             values = [str(item) for item in prediction]
         else:
             values = [str(item) for item in prediction[:, i]]
         results[key] = values
 
-    results['Probabilities'] = [str(prob) for prob in probabilities]
-    results['AD'] = [None for _ in range(len(prediction))]
+    results["Probabilities"] = [str(prob) for prob in probabilities]
+    results["AD"] = [None for _ in range(len(prediction))]
+
+    final_all = {"predictions": [dict(zip(results, t)) for t in zip(*results.values())]}
+
+    return final_all
+
+
+def graph_post_handler(request: PredictionRequestPydantic):
+    # Obtain the request info
+    onnx_model = base64.b64decode(request.model["rawModel"])
+    ort_session = onnxruntime.InferenceSession(onnx_model)
+    feat_config = request.extraConfig["torchConfig"]["featurizer"]
+    # Load the featurizer
+    featurizer = SmilesGraphFeaturizer()
+    featurizer.load_json_rep(feat_config)
+    smiles = request.dataset["input"][0]["SMILES"]
```

Comment:
No need to hardcoded the smiles here, just upload the input to the model

Reply:
fixed. Need approval for merge